Intersect.Client/Interface/Game/Map/MinimapWindow.cs

```diff
@@ -0,0 +1,571 @@
+                                {
+                                    texSet = true;
+                                }
+                            }
+                            // Is it a None tool?
+                            else if (tool == -1)
+                            {
+                                color = Options.Instance.MinimapOpts.MinimapColors.Resource["None"];
+                                colSet = true;
+                                texture = Options.Instance.MinimapOpts.MinimapImages.Resource["None"];
+                                texSet = true;
+                            }
+
+                            // Have we managed to set our color? If not, set to default.
+                            if (!colSet)
+                            {
+                                color = Options.Instance.MinimapOpts.MinimapColors.Default;
+                            }
+
+                            // Have we managed to set our texture? If not, set to blank.
+                            if (!texSet)
+                            {
+                                texture = string.Empty;
+                            }
+
+                            break;
+
+                        case EntityType.Projectile:
+                            continue;
+
+                        default:
+                            color = Options.Instance.MinimapOpts.MinimapColors.Default;
+                            texture = Options.Instance.MinimapOpts.MinimapImages.Default;
+                            break;
+                    }
+                }
+
+                // Add this to our location dictionary!
+                if (!dict.TryGetValue(map, out var locationDictionary))
+                {
+                    locationDictionary = new Dictionary<Point, EntityInfo>();
+                    dict.Add(map, locationDictionary);
+                }
+
+                if (color == null || texture == null)
+                {
+                    continue;
+                }
+
+                locationDictionary.TryAdd(new Point(entity.X, entity.Y),
+                    new EntityInfo { Color = color, Texture = texture });
+            }
+
+            return dict;
+        }
+
+        private GameRenderTexture GenerateBaseRenderTexture()
+        {
+            var sizeX = (_minimapTileSize.X * Options.Instance.MapOpts.MapWidth) * 3;
+            var sizeY = (_minimapTileSize.Y * Options.Instance.MapOpts.MapHeight) * 3;
+
+            return Graphics.Renderer.CreateRenderTexture(sizeX, sizeY);
+        }
+
+        private GameRenderTexture GenerateMapRenderTexture()
+        {
+            var sizeX = _minimapTileSize.X * Options.Instance.MapOpts.MapWidth;
+            var sizeY = _minimapTileSize.Y * Options.Instance.MapOpts.MapHeight;
+
+            return Graphics.Renderer.CreateRenderTexture(sizeX, sizeY);
+        }
```

Comment:
I really do not think these need to be two separate methods, just parameterize the multiplier.